1_prep/src/munge_netCDFs.R

```diff
@@ -1,29 +1,155 @@
+#' each cell, for each time period
+#' @param gcm_nc filename of a GCM netCDF file
 #' @param gcm_name name of one of the six GCMs
-#' @param cell_id id of a single GCM cell that is known to contain lakes
-#' @param gcm_date bracketing years of a single GCM time period
+#' @param cell_nos vector of GCM cell ids
+#' @param gcm_time_periods - the three GCM time periods, defined by their bracketing years
+#' @param burn_in length of burn-in period, in days. Used to mirror the 
+#' start of the raw meteorological data to create a burn-in period for 
+#' the model simulation.
+#' @param burn_out length of burn-out period, in days. Used to mirror the 
+#' end of the raw meteorological data to create a burn-out period for 
+#' the model simulation.
 #' @param outfile_template string representing the filepath at which to save
-#' the feather file output of the split netCDF data. The first `%s` is
-#' used as the placeholder for the `gcm_name`, the second is for the `gcm_date`,
-#' and the third is for the `cell_no`.
-#' @return names of cell/gcm/time-period specific feather files
-munge_nc_files <- function(gcm_nc, gcm_name, cell_no, gcm_date, outfile_template) {
-  # open netCDF file (gcm-specific) and pull data for that cell, for that time period
-  # TODO - actually pull time data for only one time period - pulling all for now
-  nc <- nc_open(gcm_nc)
-  time_dim <- ncvar_get(nc, "time")
-  cell_dim <- ncvar_get(nc, "gridcells")
-  data <- purrr::map_df(nc$var, function(var) {
-    ncvar_get(nc, var, start = c(1,cell_no), count = c(-1,1))
-  }) %>% mutate(time = nc.get.time.series(f = nc,time.dim.name = "time"),.before=1)
+#' the csv file output of the split netCDF data. The first `%s` is
+#' used as the placeholder for the `gcm_name`, the second is for the time period,
+#' and the third is for the data_cell_no
+#' @return names of gcm/time-period/cell-specific csv files
+munge_nc_files <- function(gcm_nc, gcm_name, cell_nos, gcm_time_periods, burn_in, burn_out, outfile_template) {
+  # Open the netCDF file for the current GCM
+  nc <- ncdfgeom::read_timeseries_dsg(gcm_nc)
+  
+  # Pull the vectors of dates
+  nc_dates <- lubridate::as_date(nc$time)
+  
+  # Generate a list of time-period specific and cell-specific files
+  # For each cell...
+  time_period_cell_files <- purrr::map(cell_nos, function(cell_no) {
+    # For each variable stored in the netCDF...
+    cell_files <- purrr::map2_dfc(nc$data_frames, names(nc$data_frames), function(var_df, var_name) {
+      # Pull the column pertaining to the current cell and rename it to the variable name
+      var_cell_long <- var_df %>%
+        select(as.character(cell_no)) %>%
+        rename(!!var_name := as.character(cell_no))
+    }) %>%
+      # Once all variables have been compiled for each cell,
+      # Add a column for time and determine the time period for each row
+      mutate(time = nc_dates,
+             time_period = case_when(
```

Comment:
I think this should be pulled into its own function, a la

```r
#' @title Assign the correct time period string to the current date
#' @description Take a date and return the appropriate time period
#' string. Currently assumes that the time periods are one of 
#' "1981_2000", "2040_2059", or "2080_2099". Could be improved later
#' to handle others.
#' @param time vector of `Date` class values
#' @param gcm_time_periods vector of the character strings representing 
#' available time periods for the data. One of "1981_2000", "2040_2059", 
#' or "2080_2099"
#' @return vector of the same length as `time` with the time period strings
#' 
date_to_time_period <- function(time, gcm_time_periods) {
  dplyr::case_when(
    time <= as.Date("2010-01-01") ~ gcm_time_periods[1],
    time <= as.Date("2070-01-01") ~ gcm_time_periods[2],
    TRUE ~ gcm_time_periods[3]
  )
}
```

Then you can use it here and in `munge_gcm_dates()` and it is a bit better solution for documenting the brittle nature of assuming `gcm_time_periods` is always the same because it is pulled out into a helper function, so we only have to fix it in one place in the future when we want to improve.

Reply:
Example using it, 

```r
> gcm_time_periods <- c("1981_2000", "2040_2059", "2080_2099")
> date_to_time_period(as.Date(c("1995-08-29", "2088-10-03", "2059-01-01")), gcm_time_periods)
[1] "1981_2000" "2080_2099" "2040_2059"
```

So you should be able to edit the above to `time_period = date_to_time_period(time, gcm_time_periods)`

Reply:
Good idea -- I'll make that change